scripts/install.js

```diff
@@ -52,15 +52,26 @@ function provideNinja() {
         stdio: [0, 1, 2],
       });
       console.log("No prebuilt Ninja, building Ninja now");
-      var build_ninja_command = "./configure.py --bootstrap";
+      var build_ninja_command = is_windows
+        ? "python ./configure.py --bootstrap --platform mingw"
+        : "./configure.py --bootstrap";
+
       child_process.execSync(build_ninja_command, {
         cwd: ninja_source_dir,
         stdio: [0, 1, 2],
       });
       fs.copyFileSync(path.join(ninja_source_dir, ninja_bin_filename), ninja_bin_output);
     } else {
       console.log(`ninja.tar.gz not availble in CI mode`);
-      require("../ninja/snapshot").build();
```

Comment:
Not sure I understand the implications of this change.
There are some other things that `ninja/snapshot.js` seem to be doing. E.g. the last commit there is [tweak to support m1](https://github.com/rescript-lang/ninja/commit/3a477ee7cac769d25f0a8993612a7ea59411e48c)

Reply:
Good point. I'll see if I can restore building via `ninja/snapshot.js`.

But why was this handled differently anyway when building from `../vendor/ninja.tar.gz` (not using snapshot.js) vs. when building from the ninja submodule (using snapshot.js)?

Maybe the first codepath is obsolete (as `vendor/ninja.tar.gz` does not actually exist)? In that case, we should probably remove it?

Reply:
The ninja build from source 
- on Win: it is running under visual studio command line
- on *nix: it is built with -lto and debugger info was stripped

Ninja is rarely changed, so I only built it on the Linux CI, and binaries for three platforms are vendored,
why it is vendored is to avoid Python toolchain

Reply:
@bobzhang Ok, then my suggestion would be to treat all three platforms the same:

* In the CI for the rescript-compiler repo, always use the vendored ninja binaries.
* Add CI to the rescript-lang/ninja repo to produce the binaries to vendor for all platforms.

What do you think?